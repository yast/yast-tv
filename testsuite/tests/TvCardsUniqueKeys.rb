# encoding: utf-8

module Yast
  class TvCardsUniqueKeysClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      # Tv::cards == nil
      Tv.cards = nil
      @keys = Tv.CardsUniqueKeys
      DUMP(@keys)

      # Tv::cards == $[]
      Tv.cards = []
      DUMP(@keys)

      # Tv::cards == card1
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        }
      ]
      @keys = Tv.CardsUniqueKeys
      DUMP(@keys)

      # Tv::cards == card1, nil, nil, card4
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        },
        nil,
        nil,
        {
          "module"        => "xyz",
          "parameters"    => { "par_other" => "testvalue2" },
          "unique_key"    => "j7F3.J8dhgs34fK2",
          "sound_card_no" => 1
        }
      ]
      @keys = Tv.CardsUniqueKeys
      DUMP(@keys)

      nil
    end
  end
end

Yast::TvCardsUniqueKeysClient.new.main
