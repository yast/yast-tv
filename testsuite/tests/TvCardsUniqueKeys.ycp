{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size" : -1 ],
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

	// Tv::cards == nil
	Tv::cards = nil;
	list keys = Tv::CardsUniqueKeys();
	DUMP(keys);

	// Tv::cards == $[]
	Tv::cards = [];
	DUMP(keys);

	// Tv::cards == card1
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ]
	    ];
	keys = Tv::CardsUniqueKeys();
	DUMP(keys);

	// Tv::cards == card1, nil, nil, card4
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ],
	      nil,
	      nil,
	      $[ "module" : "xyz",
		 "parameters" : $[ "par_other" : "testvalue2" ],
		 "unique_key" : "j7F3.J8dhgs34fK2",
		 "sound_card_no" : 1
	      ]
	    ];
	keys = Tv::CardsUniqueKeys();
	DUMP(keys);
}
