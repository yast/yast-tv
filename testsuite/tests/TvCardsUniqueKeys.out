Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp"
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system nil
Dump	[]
Dump	[]
Dump	["hluk.HvsIesmXXX1"]
Dump	["hluk.HvsIesmXXX1", "j7F3.J8dhgs34fK2"]
